node/data_node.go

```diff
@@ -0,0 +1,341 @@
+// Copyright (C) 2019-2023 Algorand, Inc.
+// This file is part of go-algorand
+//
+// go-algorand is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// go-algorand is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with go-algorand.  If not, see <https://www.gnu.org/licenses/>.
+
+// Package node is the Algorand node itself, with functions exposed to the frontend
+package node
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"path/filepath"
+	"time"
+
+	"github.com/algorand/go-algorand/agreement"
+	"github.com/algorand/go-algorand/catchup"
+	"github.com/algorand/go-algorand/config"
+	"github.com/algorand/go-algorand/data"
+	"github.com/algorand/go-algorand/data/account"
+	"github.com/algorand/go-algorand/data/basics"
+	"github.com/algorand/go-algorand/data/bookkeeping"
+	"github.com/algorand/go-algorand/data/transactions"
+	"github.com/algorand/go-algorand/ledger"
+	"github.com/algorand/go-algorand/ledger/ledgercore"
+	"github.com/algorand/go-algorand/logging"
+	"github.com/algorand/go-algorand/network"
+	"github.com/algorand/go-algorand/network/messagetracer"
+	"github.com/algorand/go-algorand/rpcs"
+	"github.com/algorand/go-algorand/stateproof"
+	"github.com/algorand/go-algorand/util/db"
+	"github.com/algorand/go-algorand/util/execpool"
+)
+
+// AlgorandDataNode implements sync mode/ledger delta APIs and disables participation-related methods
+type AlgorandDataNode struct {
```

Comment:
Let me think about it--also open to suggestions. I don't want to necessarily name it "sync node" since I think there is a broader set of use cases that generally relate to ledger data access, accessibility, and visibility.  

Things like the txn-tracer functionality that is being built will probably require launching the node w/ a different db tracker which you wouldn't want to run in a conventional node for example.

Reply:
I went with Follower for the rename. I think it implies non-participation, and also makes sense through a leader-follower lens as well.